src/mapGenerator/MapUtility.cpp

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2005 - 2015 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Return To The Roots.
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#include "mapGenerator/MapUtility.h"
+#include "mapGenerator/ObjectGenerator.h"
+#include "mapGenerator/VertexUtility.h"
+#include "gameData/TerrainData.h"
+
+#define _USE_MATH_DEFINES
+#include <math.h>
+#include <queue>
+#include <list>
+#include <vector>
+#include <algorithm>
+#include <stdlib.h>
+#include <stdexcept>
+
+void MapUtility::SetHill(Map* map, const Vec2& center, int z)
+{
+    std::vector<int> neighbors = VertexUtility::GetNeighbors(center.x, center.y, map->width, map->height, z);
+    for (std::vector<int>::iterator it = neighbors.begin(); it != neighbors.end(); ++it)
+    {
+        const int x2 = *it % map->width, y2 = *it / map->width;
+        const double d = VertexUtility::Distance(center.x, center.y, x2, y2, map->width, map->height);
+        map->z[*it] = std::max((unsigned char)(z - d), (unsigned char)map->z[*it]);
+    }
+}
+
+unsigned int MapUtility::ComputeWaterSize(Map* map, const Vec2& position, const unsigned int max)
```

Comment:
Method could be more general, like "GetBodySize(Map* map, TerrainType terrain, Vec2 pos, int max)".